source/guide_mailman-3.rst

```diff
@@ -0,0 +1,464 @@
+
+----
+
+.. note:: For this guide you should be familiar with the basic concepts of
+
+  * Python_
+  * supervisord_
+  * Folder/File Permissions
+
+License
+=======
+
+Mailman is released under the GNU General Public License
+
+  * https://www.gnu.org/copyleft/gpl.html
+
+Prerequisites
+=============
+
+Your URL needs to be setup for web and mail:
+
+::
+
+ [isabell@stardust ~]$ uberspace web domain list
+ isabell.uber.space
+ [isabell@stardust ~]$ uberspace mail domain list
+ isabell.uber.space
+ [isabell@stardust ~]$
+
+Additionally, create a mailbox_ for Mailman to use to send e-mails. In this example, we are going to use ``forwarder@isabell.uber.space``.
+
+Installation
+============
+
+Get Mailman 3
+-------------
+Install Mailman 3 and its dependencies via pip.
+
+::
+
+ [isabell@stardust ~]$ pip3.6 install --user mailman postorius hyperkitty mailman-hyperkitty django whoosh
+ [...]
+ [isabell@stardust ~]$
+
+Get Dart Sass
+-------------
+Postorius and HyperKitty are build using SASS_ stylesheets which have to be compiled/decompiled before use. As U7 does not provide an SASS compiler, we need to download the *Dart Sass* standalone binary. Get the link of the latest Dart Sass binary for **linux-x64** from https://github.com/sass/dart-sass/releases, download and extract the binaries:
+
+::
+
+ [isabell@stardust ~]$ wget https://github.com/sass/dart-sass/releases/download/9.99.9/dart-sass-9.99.9-linux-x64.tar.gz
+ [isabell@stardust ~]$ tar xzvf dart-sass-9.99.9-linux-x64.tar.gz dart-sass
+ [isabell@stardust ~]$ mv dart-sass ./bin/ 
+ [isabell@stardust ~]$ rm dart-sass-9.99.9-linux-x64.tar.gz 
+ [isabell@stardust ~]$
+
+
+Get the mailman-suite example configuration
+-------------------------------------------
+To have a starting point for configuration, we use the mailman-suite_ example provided by the Mailman team and move it to a path where we can access it easily.
+
+::
+
+ [isabell@stardust ~]$ git clone https://gitlab.com/mailman/mailman-suite.git
+ [isabell@stardust ~]$ mv mailman-suite/mailman-suite_project/ .
+ [isabell@stardust ~]$ rm -rf mailman-suite
+ [isabell@stardust ~]$ mv mailman-suite_project/ mailman-suite
+ [isabell@stardust ~]$ 
+
+
+Get and enable uwsgi
+---------------------
+.. include:: includes/install-uwsgi.rst
+
+Get .qmail helper script
+-------------------------
+Mailman 3 uses LMTP to transfer emails locally. As qmail_ is not able to use this directly, we need to download a helper script from the mailman source and adjust it to our needs:
+
+::
+
+ [isabell@stardust ~]$ cd bin
+ [isabell@stardust bin]$ wget https://gitlab.com/mailman/mailman/raw/master/contrib/qmail-lmtp
+ [isabell@stardust bin]$ chmod +x qmail-lmtp
+
+Now edit ``qmail-lmtp`` and change the two following lines:
+
+.. code :: python
+
+ 1: #!/usr/bin/env python3.6
+ [...]
+ 18:            os.environ['EXT' + (sys.argv[2] if len(sys.argv) >= 3 else '')] + "@" + os.environ['HOST'],
```

Comment:
Can you expand a bit on why this is needed? Is this only relevant to our specific setup or to all qmail installations? If all of them are affected, I think contributing this to the mailman repo would be nice.

Reply:
It is relevant for all qmail installations that want to use mailman3, see official docs: https://mailman.readthedocs.io/en/latest/src/mailman/docs/mta.html#qmail

Reply:
As mentioned, I was only following the official docs on this.
Unfortunately I was not able to use the "official" helper script in the original form as it expects an environment variable `EXT1 `(being called by ``echo '|qmail-lmtp 8042 1' > .qmail`` where the last ``1`` is used to stich the variable name of `EXT1 `together. 
When I invoked the original script via qmail, the environment variable ``EXT1`` was not found. On U7, we only have ``EXT``, ``EXT2``, ``EXT3`` and ``EXT4``. Not sure where this comes from though.

Reply:
According to [the qmail docs](http://www.qmail.org/man/man8/qmail-command.html), there is not `EXT1` :sweat_smile: did you ever encounter one in the wild and, if yes, where?

Reply:
No, actually I didn't. As said before, just followed the official docs. 
Checked the qmail docs as well yesterday and will create a pull request for the `qmail-lmtp` script in the official repository so we can fully comply with their documentation.